packages/core/src/modules/discover-features/FeatureRegistry.ts

```diff
@@ -0,0 +1,52 @@
+import type { FeatureQuery, Feature } from './models'
+
+import { injectable } from '../../plugins'
+
+@injectable()
+class FeatureRegistry {
+  private features: Feature[] = []
+
+  /**
+   * Register a single or set of Features on the registry
+   *
+   * @param features set of {Feature} objects
+   */
+  public register(...features: Feature[]) {
+    for (const feature of features) {
+      const index = this.features.findIndex((item) => item.type === feature.type && item.id === feature.id)
+
+      if (index > -1) {
+        this.features[index] = this.features[index].combine(feature)
```

Comment:
I like the combine approach 👍 . Maybe we can add an optional override feature later that will not combine but override